Update examples before upmerge (#4)
ArekBalysNordic/sdk-nrf: samples/matter/light_switch/README.rst

   :local:
   :depth: 2

> This light switch sample demonstrates the usage of the :ref:`Matter <ug_matter>` (formerly Project Connected Home over IP, Project CHIP) application layer to build a switch device that bind with lighting devices and change the state of their LEDs.

Reviewer: ```suggestion
This light switch sample demonstrates the usage of the :ref:`Matter <ug_matter>` (formerly Project Connected Home over IP, Project CHIP) application layer to build a switch device that binds with lighting devices and changes the state of their LEDs.
```

   :header: heading
   :rows: nrf52840dk_nrf52840, nrf5340dk_nrf5340_cpuapp, nrf21540dk_nrf52840

For this sample to work, you also need at least one :ref:`Matter light bulb <matter_light_bulb_sample>` sample programmed to another supported development kit.

To commission device and do all commands needed to prepare all device to work you need also a controller. 

> You can use a `Chip-Tool <https://github.com/project-chip/connectedhomeip/tree/master/docs/guides/chip_tool_guide.md>`_ to perform commissioning and prepare all devices to work.

Reviewer 2: The doc still doesn't exist, so I suggest removing it and adding when it's available.

To enable the Matter CLI, you must build the light switch application with the additional option ``-DCONFIG_CHIP_LIB_SHELL=y``.
Use the following command pattern, with ``<build-target>`` replaced with the build target name of Nordic Semiconductor's development kit you are using (see `Requirements`_):

.. parsed-literal::
   :class: highlight

   west build -b <build-target> -- -DCONFIG_CHIP_LIB_SHELL=y

For example:

.. parsed-literal::
   :class: highlight

   west build -b nrf52840dk_nrf52840_cpuapp -- -DCONFIG_CHIP_LIB_SHELL=y

Device Firmware Upgrade support
===============================

.. include:: ../lock/README.rst
    :start-after: matter_door_lock_sample_build_with_dfu_start
    :end-before: matter_door_lock_sample_build_with_dfu_end

.. _matter_light_switch_sample_ui:

User interface
**************

.. include:: ../lock/README.rst
    :start-after: matter_door_lock_sample_led1_start
    :end-before: matter_door_lock_sample_led1_end

LED 2:
   Shows the state of the Bluetooth LE Direct Firmware Update (DFU) process.

   * Off - Bluetooth LE is not advertising and DFU is not ready.

>    * Rapid Even Flashing (30 ms off / 170 ms on) - Bluetooth LE is advertising and the DFU process can be started .

Reviewer: ```suggestion
   * Rapid Even Flashing (30 ms off / 170 ms on) - Bluetooth LE is advertising and the DFU process can be started.
```

      File not found: ./ncs/nrf/samples/matter/light_switch/configuration/nrf52840dk_nrf52840/prj_shell.conf

Testing
*******

After building the sample and programming it to your development kit, complete the steps in the following sections.

.. _matter_light_switch_sample_prepare_for_testing:

Prepare for testing
===================

After building this and the :ref:`Matter Light Bulb <matter_light_bulb_sample>` samples, and programming them to the development kits, complete the following steps:

.. note::

>    In both examples (light switch and light bulb), a Bluetooth LE discriminator is set with the same value by default (in hex 0xF00 in decimal 3840).

Reviewer: ```suggestion
   In both samples (light switch and light bulb), a Bluetooth LE discriminator is set with the same value by default (hexidecimal: ``0xF00``; decimal: ``3840``).
```

>    That means before commissioning, only one uncommissioned device can be powered up; otherwise, chip-tool can commission a random device.

Reviewer: ```suggestion
   This means that only one uncommissioned device can be powered up before commissioning.
   If both are powered up at the same time, the CHIP Tool can commission a random device and the node ID assignment is also random.
```

>    If a device becomes commissioned, you can power up next device and perform the subsequent commissioning.

Reviewer: ```suggestion
   When one device is commissioned, power up the next device and perform the commissioning.
```

>    To avoid this unclear situation you can set up your unique discriminator in *src/chip_project_config.h* file by changing `CHIP_DEVICE_CONFIG_USE_TEST_SETUP_DISCRIMINATOR` value.

Reviewer: ```suggestion
   To avoid this unclear situation, you can set up your unique discriminator in :file:`src/chip_project_config.h` file by changing :kconfig:option:`CHIP_DEVICE_CONFIG_USE_TEST_SETUP_DISCRIMINATOR` value.
```
